src/cache.jl

```diff
@@ -229,24 +246,27 @@ function fit_nuisance!(cache::TMLECache; verbosity=1)
 end
 
 
-function tmle_step!(cache::TMLECache; verbosity=1, threshold=1e-8)
+function tmle_step!(cache::TMLECache; verbosity=1, threshold=1e-8, weighted_fluctuation=false)
     # Fit fluctuation
     offset = TMLE.compute_offset(cache.data[:Q₀])
     W = TMLE.confounders(cache.data[:no_missing], cache.Ψ)
     jointT = TMLE.joint_treatment(TMLE.treatments(cache.data[:no_missing], cache.Ψ))
-    covariate = TMLE.compute_covariate(jointT, W, cache.η.G, cache.data[:indicators_str]; threshold=threshold)
+    covariate, w = TMLE.covariate_and_weights(
+        jointT, W, cache.η.G, cache.data[:indicators_str]; 
+        threshold=threshold, weighted_fluctuation=weighted_fluctuation
+    )
     X = TMLE.fluctuation_input(covariate, offset)
     y = TMLE.target(cache.data[:no_missing], cache.Ψ)
-    mach = machine(cache.η_spec.F, X, y, cache=cache.η_spec.cache)
+    mach = machine(cache.η_spec.F, X, y, w, cache=cache.η_spec.cache)
```

Comment:
For clarity, perhaps it would be good to call the weight, "weight" rather than "w" (to not confuse it with the variable W in the causal graph that here appears as part of X). No issue with code, but better naming to make clear this is the weight factor.